save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/ProjectView.kt

```diff
@@ -5,137 +5,407 @@
+                it.set("Content-Type", "application/json")
+            }
+            project = get("${window.location.origin}/getProject?name=${props.name}&owner=${props.owner}", headers)
+                .json()
+                .await()
+                .unsafeCast<Project>()
+            val jsonProject = JSON.stringify(project)
+            gitDto = post("${window.location.origin}/getGit", headers, jsonProject)
+                .json()
+                .await()
+                .unsafeCast<GitDto?>()
+            setState { isLoading = false }
+        }
     }
 
+    @Suppress("ComplexMethod", "TOO_LONG_FUNCTION")
     private fun submitExecutionRequest() {
-        val executionRequest = ExecutionRequest(
-            project = props.executionRequest.project,
-            gitRepository = props.executionRequest.gitRepository,
-            propertiesRelativePath = pathToPropertyFile ?: "save.properties"
-        )
+        if (numberOpenningCard == 0) {
+            setState {
+                isErrorOpen = true
+                errorLabel = "No project type"
+                errorMessage = "Please choose one of the project types"
+            }
+        } else if (numberOpenningCard == 1) {
+            gitUrlProject?.let {
+                val newGitDto = GitDto(url = it)
+                submitExecutionRequestGit(newGitDto)
+            } ?: gitDto?.let {
+                submitExecutionRequestGit(it)
+            } ?: setState {
+                isErrorOpen = true
+                errorLabel = "No git url"
+                errorMessage = "Please provide a git url"
+            }
+        } else {
+            if (selectedTypes.isEmpty()) {
+                setState {
+                    isErrorOpen = true
+                    errorLabel = "Both type of project"
+                    errorMessage = "Please choose one of type test suites"
+                }
+                return
+            }
+            state.binaryFile ?: run {
+                setState {
+                    isErrorOpen = true
+                    errorLabel = "No binary file has been selected"
+                    errorMessage = "Please select binary file"
+                }
+                return
+            }
+            state.propertyFile ?: run {
+                setState {
+                    isErrorOpen = true
+                    errorLabel = "No property file has been selected"
+                    errorMessage = "Please select property file"
+                }
+                return
+            }
+            submitExecutionRequestBinFile()
+        }
+    }
+
+    private fun submitExecutionRequestBinFile() {
+        val headers = Headers()
+        val formData = FormData()
+        val request = ExecutionRequestForStandardSuites(project, selectedTypes)
+        formData.append("execution", Blob(arrayOf(JSON.stringify(request)), BlobPropertyBag("application/json")))
+        state.propertyFile?.let { formData.append("property", it) }
```

Comment:
Isn't it guaranteed that both files are present at this point? Also, it would be an error to silently ignore their nullability and still send the request.

Reply:
In `submitExecutionRequest` I've written handler for this case